### PR TITLE
Add fields for ML Storage CRD

### PR DESCRIPTION
**Pull request by nikita-vanyasin**
- copy/move a few security-related constants into shared package

### CODE REVIEW

**Review by jwierzbo — APPROVED** [review]

File: pkg/apis/ml/v1alpha1/storage_s3_spec.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+package v1alpha1
+
+import (
+	"github.com/pkg/errors"
+)
+
+type ArangoMLStorageS3Spec struct {
+	// Endpoint specifies the S3 API-compatible endpoint which implements storage
+	// Required
+	Endpoint string `json:"endpoint"`
```
Comment (ajanikow) [inline pkg/apis/ml/v1alpha1/storage_s3_spec.go:30]: Use Pointers and Get functions (which return default values)

File: pkg/apis/ml/v1alpha1/storage_spec.go [hunk 2]
```diff
@@ -20,5 +20,48 @@
...
+
 type ArangoMLStorageSpec struct {
+	// ListenPort defines on which port the sidecar container will be listening for connections
+	// +doc/default: 9201
+	ListenPort *uint16 `json:"listenPort,omitempty"`
+
+	// Resources holds resource requests & limits for container running the S3 proxy
+	// +doc/type: core.ResourceRequirements
+	// +doc/link: Documentation of core.ResourceRequirements|https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#resourcerequirements-v1-core
+	Resources core.ResourceRequirements `json:"resources,omitempty"`
```
Comment (ajanikow) [inline pkg/apis/ml/v1alpha1/storage_spec.go:37]: Put it under structure for mode, like Mode.Sidecar

File: pkg/apis/ml/v1alpha1/storage_spec.go [hunk 3]
```diff
@@ -20,5 +20,48 @@
...
+func (s *ArangoMLStorageSpec) Validate() error {
+	if s.S3 == nil {
+		return errors.New("Currently only s3 storage type is supported")
+	}
+
+	return s.S3.Validate()
+}
+
+// SetDefaults fills in missing defaults
+func (s *ArangoMLStorageSpec) SetDefaults() {
```
Comment (ajanikow) [inline pkg/apis/ml/v1alpha1/storage_spec.go:51]: Remove it, instead return defaults in GetX

File: pkg/apis/ml/v1alpha1/storage_spec.go [hunk 4]
```diff
@@ -20,5 +20,48 @@
...
+
+	// Resources holds resource requests & limits for container running the S3 proxy
+	// +doc/type: core.ResourceRequirements
+	// +doc/link: Documentation of core.ResourceRequirements|https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#resourcerequirements-v1-core
+	Resources core.ResourceRequirements `json:"resources,omitempty"`
+
+	S3 *ArangoMLStorageS3Spec `json:"s3,omitempty"`
+}
+
+func (s *ArangoMLStorageSpec) Validate() error {
```
Comment (ajanikow) [inline pkg/apis/ml/v1alpha1/storage_spec.go:42]: Missing path

File: pkg/apis/ml/v1alpha1/storage_spec.go [hunk 5]
```diff
@@ -20,5 +20,48 @@
...
+	// ListenPort defines on which port the sidecar container will be listening for connections
+	// +doc/default: 9201
+	ListenPort *uint16 `json:"listenPort,omitempty"`
+
+	// Resources holds resource requests & limits for container running the S3 proxy
+	// +doc/type: core.ResourceRequirements
+	// +doc/link: Documentation of core.ResourceRequirements|https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#resourcerequirements-v1-core
+	Resources core.ResourceRequirements `json:"resources,omitempty"`
+
+	S3 *ArangoMLStorageS3Spec `json:"s3,omitempty"`
```
Comment (ajanikow) [inline pkg/apis/ml/v1alpha1/storage_spec.go:39]: Add it under section Backend

**Pull request merged**